node/internal.ts

```diff
@@ -303,12 +305,12 @@ export function _command(command: string, properties: any, message: string) {
     _writeLine(taskCmd.toString());
 }
 
-export function _warning(message: string, source?: IssueSource): void {
-    _command('task.issue', { 'type': 'warning', 'source': source }, message);
+export function _warning(message: string, source: IssueSource = IssueSource.TaskInternal): void {
+    _command('task.issue', { 'type': 'warning', 'source': source, 'token': _taskSdkToken }, message);
 }
 
-export function _error(message: string, source?: IssueSource): void {
-    _command('task.issue', { 'type': 'error', 'source': source }, message);
+export function _error(message: string, source: IssueSource = IssueSource.TaskInternal): void {
```

Comment:
what if a non-built in task calls _error.  is IssueSource ignored?

I'm thinking a better default for this would be "Unspecified" rather than TaskInternal or CustomerSource.  Then we can deal with the default behavior upstream.  Otherwise, changing the default behavior will require more churn.

I anticipate we'll want to change the default later...  I understand choosing "TaskInternal" for now to reduce churn.  However, we may want to change the default to opt-in.

Reply:
I think we can keep it as it is in the task-lib repo. It's easy to implement the behavior you've described on the agent side. Because the agent knows if the task is in-the-box or not but the task itself doesn't.